src/app/jscaip/DIRECTION.ts

```diff
@@ -121,6 +124,26 @@ export class Direction {
         }
         throw new Error("Invalid argument for Direction (y = " + y + ")")
     }
+    public static fromMove(start: Coord, end: Coord): Direction {
+        if (start.x === end.x && start.y === end.y) {
+            throw new Error("Invalid direction from static move");
+        } else if (start.x === end.x) {
+            // Vertical move
+            if (start.y < end.y) return Direction.DOWN;
+            if (start.y > end.y) return Direction.UP;
+        } else if (start.y === end.y) {
+            // Horizontal move
+            if (start.x < end.x) return Direction.RIGHT;
+            if (start.y > end.y) return Direction.LEFT;
+        } else if (Math.abs(start.x - end.x) === Math.abs(start.y - end.y)) {
+            // Diagonal move
+            if (start.x < end.x && start.y < end.y) return Direction.DOWN_RIGHT;
+            if (start.x < end.x && start.y > end.y) return Direction.UP_RIGHT;
+            if (start.x > end.x && start.y < end.y) return Direction.DOWN_LEFT;
+            if (start.x > end.x && start.y > end.y) return Direction.UP_LEFT;
+        }
+        throw new Error("Invalid direction from move: " + start.toString() + "->" + end.toString());
+    }

```

Comment:
C'est bizarre, je croyais avoir fait un truc du genre.
En appelant of(delta.x, delta.y) cela ne ferait-il pas le même résultat ?

Reply:
`Direction.of` fait quelque chose de similaire, mais seulement pour une distance de 1 (alors qu'ici c'est adapté à toute distance >= 1). Peut-être que ceci devrait être déplacé dans `Direction.of`.

Reply:
Ouais, centralisons ça dans la classe ou ça a le plus de sens ?